virtool/analyses/api.py

```diff
@@ -333,116 +351,136 @@ async def upload(req: Request) -> Response:
+            404: Not found
+        """
+        pg = self.request.app["pg"]
+        upload_id = int(self.request.match_info["upload_id"])
 
-    if not analysis_file:
-        raise NotFound()
+        analysis_file = await get_row_by_id(pg, AnalysisFile, upload_id)
 
-    analysis_file_path = (
-        req.app["config"].data_path / "analyses" / analysis_file.name_on_disk
-    )
+        if not analysis_file:
+            raise NotFound()
 
-    if not analysis_file_path.exists():
-        raise NotFound("Uploaded file not found at expected location")
+        analysis_file_path = (
+                self.request.app["config"].data_path / "analyses" / analysis_file.name_on_disk
+        )
 
-    return FileResponse(analysis_file_path)
+        if not analysis_file_path.exists():
+            raise NotFound("Uploaded file not found at expected location")
 
+        return FileResponse(analysis_file_path)
 
-@routes.get("/analyses/documents/{analysis_id}.{extension}")
-async def download_analysis_document(req: Request) -> Response:
-    """
-    Download an analysis document.
 
-    """
-    db = req.app["db"]
+@routes.view("/analyses/documents/{analysis_id}.{extension}")
+class DocumentDownloadView(PydanticView):
 
-    analysis_id = req.match_info["analysis_id"]
-    extension = req.match_info["extension"]
+    async def get(self) -> Union[r200[Response], r404]:
+        """
+        Download an analysis document.
 
-    document = await db.analyses.find_one(analysis_id)
+        Status Codes:
+            200: Operation successful
+            404: Not found
+        """
+        db = self.request.app["db"]
 
-    if not document:
-        raise NotFound()
+        analysis_id = self.request.match_info["analysis_id"]
+        extension = self.request.match_info["extension"]
 
-    if extension == "xlsx":
-        formatted = await virtool.analyses.format.format_analysis_to_excel(
-            req.app, document
-        )
-        content_type = (
-            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
-        )
-    else:
-        formatted = await virtool.analyses.format.format_analysis_to_csv(
-            req.app, document
-        )
-        content_type = "text/csv"
+        document = await db.analyses.find_one(analysis_id)
 
-    headers = {
-        "Content-Disposition": f"attachment; filename={analysis_id}.{extension}",
-        "Content-Type": content_type,
-    }
+        if not document:
+            raise NotFound()
+
+        if extension == "xlsx":
+            formatted = await virtool.analyses.format.format_analysis_to_excel(
+                self.request.app, document
+            )
+            content_type = (
+                "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+            )
+        else:
+            formatted = await virtool.analyses.format.format_analysis_to_csv(
+                self.request.app, document
+            )
+            content_type = "text/csv"
```

Comment:
Can you update this such that a `400` with a message like `f"Invalid extension: {extension}"` is returned when the extension is not `xlsx` or `csv`?

Make sure you update the status codes in the docstring to match.